tests/testApplyLookupTable.py

```diff
@@ -0,0 +1,84 @@
+import unittest
+
+import numpy as np
+
+import lsst.utils.tests
+import lsst.afw.geom as afwGeom
+import lsst.afw.image as afwImage
+from lsst.ip.isr import applyLookupTable
+
+def referenceApply(image, table, indOffset):
+    """!Reference implementation of applyLookupTable
+
+    The algorithm is as follows:
+        numOutOfRange = 0
+        For each i,j of the image:
+            lookupInd = int(indOffset + image[i,j])
+            if lookupInd not in range [0, table.size() - 1]:
+                set lookupInd to nearest edge and increment numOutOfRange
+            image[i,j] += table[lookupInd]
+        return numOutOfRange
+
+    @param[in,out] image  image to which to add the values; modified in place
+    @param[in] table  lookup table
+    @param[in] indOffset  scalar added to image value before truncating to lookup column
+
+    @return the number of pixels whose values were out of range
+    """
+    imArr = image.getArray()
+    indArr = np.array(imArr + indOffset, dtype=int)
+    maxInd = len(table) - 1
+    numBadPoints = np.sum(indArr < 0)
+    numBadPoints += np.sum(indArr > maxInd)
+    indArr = np.where(indArr < 0, 0, indArr)
+    indArr = np.where(indArr >= maxInd, maxInd, indArr)
+    imArr += table[indArr]
+    return numBadPoints
```

Comment:
Is there a reason not to use this as the actual implementation?  We should prefer a python implementation unless there's demonstrated performance advantage to having it in C++.  Given that there's a simple python implementation, I think you need to demonstrate that it's not too slow before we should accept the C++.